lib/rust/ensogl/component/grid-view/src/header.rs

```diff
@@ -0,0 +1,655 @@
+//! The module containing [`GridView`] with headers.
+
+use crate::prelude::*;
+
+use crate::entry;
+use crate::entry::visible::VisibleEntry;
+use crate::visible_area;
+use crate::Col;
+use crate::Entry;
+use crate::Properties;
+use crate::Row;
+
+use ensogl_core::application::Application;
+use ensogl_core::display;
+use ensogl_core::display::scene::layer::WeakLayer;
+use ensogl_core::display::scene::Layer;
+use ensogl_scroll_area::Viewport;
+
+
+
+// ===========
+// === FRP ===
+// ===========
+
+/// A structure with layers where the headers are displayed. Designed to be used in FRP networks.
+///
+/// To avoid unpredictable layer lifetime, all the references are weak.
+#[allow(missing_docs)]
+#[derive(Clone, Debug, Default)]
+pub struct WeakLayers {
+    pub header: Option<WeakLayer>,
+    pub text:   Option<WeakLayer>,
+}
+
+impl WeakLayers {
+    /// Constructor.
+    pub fn new(header: &Layer, text: Option<&Layer>) -> Self {
+        Self { header: Some(header.downgrade()), text: text.map(|l| l.downgrade()) }
+    }
+
+    /// Upgrade the main layer for headers.
+    pub fn upgrade_header(&self) -> Option<Layer> {
+        self.header.as_ref()?.upgrade()
+    }
+
+    /// Upgrade the text layer for headers.
+    pub fn upgrade_text(&self) -> Option<Layer> {
+        self.text.as_ref()?.upgrade()
+    }
+}
+
+ensogl_core::define_endpoints_2! { <HeaderModel: (frp::node::Data)>
+    Input {
+        set_layers(WeakLayers),
+        section_info(Range<Row>, Col, HeaderModel),
+        reset_sections(),
+    }
+    Output {
+        section_info_needed(Row, Col),
```

Comment:
`section_info_needed` doesn't fell quite self-explanatory. Is this a request for an update? When is this emitted?